Packs/MongoDBAtlas/Integrations/MongoDBAtlas/MongoDBAtlas.py

```diff
@@ -0,0 +1,586 @@
+    """
+    if not date1:
+        return date2
+    if not date2:
+        return date1
+
+    dt1 = datetime.strptime(date1, DATE_FORMAT)
+    dt2 = datetime.strptime(date2, DATE_FORMAT)
+
+    return date1 if dt1 >= dt2 else date2
+
+
+def get_last_page_of_events(client: Client, results: dict) -> dict:
+    """
+    Iterates through paginated event data, following "next" links provided in the API response until it reaches the last page.
+
+    Args:
+        client (Client): The API client instance used to retrieve the event data.
+        results (dict): The initial response dictionary containing event data and pagination links.
+
+    Returns:
+        dict: The final page of events retrieved from the API.
+    """
+    links = results.get('links', [])
+    next_url = get_page_url(links, page_type="next")
+    last_response = results
+
+    while next_url:
+        last_response = client.get_response_from_page_link(next_url)
+        links = last_response.get('links')
+        next_url = get_page_url(links, page_type='next')
+
+    return last_response
+
+
+def save_events_ids_with_specific_created_date(events: list, created_date: str) -> list:
+    """
+    Filters event IDs by a given creation date.
+
+    Args:
+        events (list): List of event dictionaries with "id" and "created" date fields.
+        created_date (str): Target creation date to filter by.
+
+    Returns:
+        list: IDs of events matching the specified creation date.
+    """
+    results = []
+    for event in events:
+        if event.get('created') == created_date:
+            results.append(event.get('id'))
+    return results
+
+
+def create_last_run_dict_for_events(output: list, new_min_time: str) -> dict:
+    """
+    Creates a dictionary to store the last run information for events.
+
+    Args:
+        output (list): List of event dictionaries.
+        new_min_time (str): The minimum time to filter events by.
+
+    Returns:
+        dict: A dictionary with `min_time` and `events_with_created_min_time` keys.
+    """
+    events_with_created_min_time = save_events_ids_with_specific_created_date(output, new_min_time)
+    return {'min_time': new_min_time,
+            'events_with_created_min_time': events_with_created_min_time
+            }
+
+
+def first_time_fetching_events(client: Client, fetch_limit: int):
+    """
+    Fetches and enriches the first batch of events, returning them with the minimum creation time.
+
+    Args:
+        client (Client): The client instance used to retrieve events.
+        fetch_limit (int): The maximum number of events to retrieve.
+
+    Returns:
+        tuple: A tuple containing:
+            - results (list): A list of event dictionaries.
+            - new_min_time (str): The creation time of the latest fetched event.
+    """
+    results = client.get_events_first_time_events(fetch_limit)
+    for event in results:
+        enrich_event(event, event_type='events')
```

Comment:
You're doing it in the `fetch_events` command as well.

Reply:
I can do it when the whole output is ready, but its another loop instead of enriching each event in an existing loop. Is that what you mean?